packages/babel-plugin-minify-dead-code-elimination/src/index.js

```diff
@@ -100,17 +105,71 @@ module.exports = ({ types: t, traverse }) => {
         }
 
         const { scope } = path;
+
+        // if the scope is created by a function, we obtain its
+        // parameter list
+        const paramsList = path.isFunction() ? path.get("params") : [];
+
+        for (let i = paramsList.length - 1; i >= 0; i--) {
+          const param = paramsList[i];
+
+          if (param.isIdentifier()) {
+            const binding = scope.bindings[param.node.name];
+            if (binding.referenced) {
+              // when the first binding is referenced (right to left)
+              // exit without marking anything after this
+              break;
+            }
+
+            Object.defineProperty(binding, markForRemoval, {
```

Comment:
Do we need to `defineProperty` here instead of just assigning it? Is this because we don't want to taint binding with "irrelevant" enumerable prop? I'm just worried about performance.